### DISCUSSION TITLE
Is it possible to use also markdonw in template?

### DISCUSSION THREAD

**Discussion by aborruso** [mkdocs/mkdocs#2514] [Q&A]
Hi,
in example is there a way to put markdown in a template file and to have it rendered as HTML?

Thank you

```
{% block content %}
  ## This is markdown
{% endblock %}
```

**Comment by Andre601** [comment]
Not sure if that is doable, but maybe add `| markdownify` to the block?

I.e.
```
{% block content | markdownify %}
  ## This markdown
{% endblock %}
```

**Reply by aborruso** [reply]
Thank you. I have

```
jinja2.exceptions.TemplateSyntaxError: expected token 'end of statement block', got '|'
```

**Reply by pawamoy** [reply]
I don't think you can use filters on blocks directly. Try

```jinja2
{% block content %}
  {% filter  markdownify %}
  ## This markdown
  {% endfilter %}
{% endblock %}
```

**Reply by aborruso** [reply]
@pawamoy I have

```
jinja2.exceptions.TemplateAssertionError: no filter named 'markdownify'
```

**Reply by pawamoy** [reply]
Then you need a Jinja extension such as https://github.com/jpsca/jinja-markdown

**Reply by aborruso** [reply]
@pawamoy I have installed it via pip. What do I must write in mkdocs.yml to enable it?

**Reply by pawamoy** [reply]
I guess you also need a plugin such as mkdocs-macros-plugin. See https://mkdocs-macros-plugin.readthedocs.io/en/latest/macros.html#the-define_env-function.

```python
from jinja_markdown import MarkdownExtension


def define_env(env):
    env.add_extension(MarkdownExtension)
```

**Reply by aborruso** [reply]
I have installed mkdocs-macros-plugin and jinja-markdown. I have added  thsi

```
{% block content %}
  {% filter  markdownify %}
  ## This markdown
  {% endfilter %}
{% endblock %}
```

WHen I serve it, I have

```
INFO     -  Building documentation...
ERROR    -  Config value: 'plugins'. Error: The "jinjamarkdown" plugin is not installed
Aborted with 1 Configuration Errors!
```

**Reply by aborruso** [reply]
@pawamoy I must correct myself. When I serve it, I have

```
jinja2.exceptions.TemplateAssertionError: no filter named 'markdownify'
```

**Reply by pawamoy** [reply]
https://github.com/jpsca/jinja-markdown uses `{% markdown %}` and `{% endmarkdown %}` block tags, not a `markdownify` filter.

**Reply by aborruso** [reply]
If I write

```
{% block content %}

{% markdown %}
# Heading

{% endmarkdown %}

{% endblock %}
```

I have

```
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'markdown'. Jinja was looking for the following tags: 'endblock'. The innermost block that needs to be closed is 'block'.
```

**Reply by pawamoy** [reply]
You need to configure mkdocs-macros-plugin to use your module containing the snippet I posted above that adds the extension to your Jinja env (long sentence, sorry 😅)

**Reply by aborruso** [reply]
I am not able to do that. I have to study, I don't want to waste any more time. You have been very kind and patient, thank you

**Reply by aborruso** [reply]
@pawamoy  I specify better: I don't want to waste your time.

**Reply by twardoch** [reply]
I made a plugin myself

https://github.com/twardoch/mkdocs-markdown-in-template-plugin
